cpp/ql/src/Security/CWE/CWE-089/SqlTainted.ql

```diff
@@ -30,7 +30,15 @@ class Configuration extends TaintTrackingConfiguration {
   }
 
   override predicate isBarrier(Expr e) {
-    super.isBarrier(e) or e.getUnspecifiedType() instanceof IntegralType
+    super.isBarrier(e)
+    or
+    e.getUnspecifiedType() instanceof IntegralType
+    or
+    exists(SqlEscapeFunction sql, int arg, FunctionInput input |
+      e = sql.getACallToThisFunction().getArgument(arg) and
+      input.isParameterDeref(arg) and
+      sql.escapesSqlArgument(input, _)
+    )
```

Comment:
I notice that `SqlPqxxTainted.ql` uses `semmle.code.cpp.dataflow.TaintTracking` whereas `SqlTainted.ql` uses the older `semmle.code.cpp.security.TaintTracking`.  Do you think it's worth upgrading it?

Reply:
Good point. Indeed, seeing as we want to extend our SQL injection coverage, I think `SqlTainted.ql` should be ported to use `semmle.code.cpp.dataflow.TaintTracking` (along with any other Code Scanning queries that use `semmle.code.cpp.security.TaintTracking`).

Reply:
On second thought, this might actually be a part of @rdmarsh2's work on https://github.com/github/codeql-c-team/issues/659 (although, I'm not sure).

Reply:
(I'm happy either way.  It doesn't need to be done right now)